src/Decompiler/Loading/Loader.cs

```diff
@@ -298,6 +309,30 @@ public Program LoadRawImage(ImageLocation imageLocation, byte[] image, Address?
             return program;
         }
 
+        private byte[]? ExtractRawFileFromArchives(ImageLocation imageLocation, byte[] image, Address? addrLoad)
+        {
+            var fsLocation = ImageLocation.FromUri(imageLocation.FilesystemPath);
+            var binaryImage = LoadBinaryImage(fsLocation, image, null, addrLoad);
+            for (int i = 0; i < imageLocation.Fragments.Length; ++i)
+            {
+                var fragment = imageLocation.Fragments[i];
+                if (binaryImage is not IArchive archive)
+                    throw new InvalidOperationException($"Image '{imageLocation}' expects an archive or file container, but none was loaded.");
+                var item = archive[fragment];
+                if (item is not ArchivedFile file)
+                    throw new InvalidOperationException($"Fragment '{fragment}' does not refer to an archived file.");
```

Comment:
We can return `file.GetBytes()` if it is last fragment
```
if (i == imageLocation.Fragments.Length - 1)
    return file.GetBytes()
```

Reply:
I'm not sure I understand these two comments. Could you perhaps illustrate with some code?

Reply:
This method should return `bytes` array from specified location. If we have `ArchivedFile` and it is last fragment then we can obtain `bytes` using `GetBytes` and it's not required to call `LoadImage`. The code will be
```
if (i == imageLocation.Fragments.Length - 1)
    return file.GetBytes()
binaryImage = file.LoadImage(Services, addrLoad);
if (binaryImage is Blob blob)
{
    // Archive doesn't know this format, perhaps Reko does.
    binaryImage = LoadBinaryImage(blob.Location, blob.Image, null, addrLoad);
}
```

Reply:
Thanks, now I understand. Changed accordingly.